resources/assets/components/utilities/CampaignProgressBar/CampaignProgressBar.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { get } from 'lodash';
+import gql from 'graphql-tag';
+import PropTypes from 'prop-types';
+import { useQuery } from '@apollo/react-hooks';
+
+import ProgressBar from '../ProgressBar/ProgressBar';
+import Spinner from '../../artifacts/Spinner/Spinner';
+import ErrorBlock from '../../blocks/ErrorBlock/ErrorBlock';
+import { getGoalInfo } from '../../../helpers/voter-registration';
+
+const CAMPAIGN_PROGRESS_BAR_QUERY = gql`
+  query CampaignProgressBarQuery($id: Int!) {
+    action(id: $id) {
+      id
+      impactGoal
+      currentImpactQuantity
+      noun
+      verb
+    }
+  }
+`;
+
+const CampaignProgressBar = ({ actionId }) => {
+  const { loading, error, data } = useQuery(CAMPAIGN_PROGRESS_BAR_QUERY, {
+    variables: { id: actionId },
+  });
+
+  const progressGoalCalculator = (currentTotal, overallGoal = 0) => {
+    if (!overallGoal || currentTotal >= overallGoal) {
+      return Math.ceil(currentTotal / 1000) * 1000;
```

Comment:
I want to walk myself through an example here in order to review this part. Say current total is `1256`. Divide that by `1000` and get `1.256`. Take the ceiling of that and get `2`. Multiply that by `1000` to get `2000`. Perfect!

Reply:
This is maybe too much of an edge case, BUT if the `currentTotal` is `0` we'll end up setting a goal of `0` (because `Math.ceil(0)` is `0`). Maybe we want to default to returning `1000` as the smallest possible goal if there isn't one set?

Reply:
🤔  good point @katiecrane so would you set one conditional for if there is no overallGoal && currentTotal is 0 and return `1000` and then keep this conditional for returning the Math.ceil()?

Reply:
Yeah! I think you could do something like this: 
```suggestion
  const progressGoalCalculator = (currentTotal, overallGoal = 0) => {
    if (!overallGoal || currentTotal >= overallGoal) {
      if (currentTotal < 1) {
        return 1000;
      }

      return Math.ceil(currentTotal / 1000) * 1000;
```

Reply:
That one I think mightttt be worth adding in (not necessarily exactly how I wrote it there are many ways you could choose to write it!) because I could see `currentTotal` being `0` right when a campaign launches.

Another edge case is if `currentTotal` is a multiple of `1000` we'll get a goal that is the same as the current total. That one seems not worth worrying about for now.